src/main/kotlin/com/zepben/evolve/streaming/grpc/Connect.kt

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2022 Zeppelin Bend Pty Ltd
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+package com.zepben.evolve.streaming.grpc
+
+import com.zepben.auth.client.ZepbenTokenFetcher
+import com.zepben.auth.client.createTokenFetcher
+import com.zepben.auth.common.AuthMethod
+import java.io.File
+
+/**
+ * A collection of functions that return a channel that connects to Evolve's gRPC service, given address and authentication details.
+ */
+object Connect {
+
+    fun connectInsecure(
+        host: String = "localhost",
+        rpcPort: Int = 50051
+    ): GrpcChannel =
+        GrpcChannelBuilder().forAddress(host, rpcPort).build()
+
+    fun connectTls(
+        host: String = "localhost",
+        rpcPort: Int = 50051,
+        ca: File? = null
+    ): GrpcChannel =
+        GrpcChannelBuilder().forAddress(host, rpcPort).makeSecure(rootCertificates = ca).build()
+
+    fun connectWithSecret(
+        clientId: String,
+        clientSecret: String,
+        confAddress: String,
```

Comment:
issue: in the python version we are taking a conf_path that by default is null - this is because the ZepbenTokenFetcher will try some default paths.
suggestion: make this similar to the python SDK and make confPath default to null, and do this for all functions with a confAddress

Reply:
`create_token_fetcher` mislabels host, path, and port as the address of the gRPC service. Rather, it is the address of the OAuth config. the `connect` methods in the python sdk assume that the host for the gRPC service and the OAuth config are the same, so only one CA is needed for both fetching the auth conf and making RPC calls. Is this correct?

Reply:
ah yep... that's originally why I specified it as conf_address :joy:

I think in that case what we should do is keep confAddress, but make it default to null, and still try the host+port default with paths of /auth and /ewb/auth (also maybe we should try /api/config/auth as well for EAS). Then if confAddress is specified we use that instead as a full address. We should also update the python SDK to match this behaviour.

The reasoning is that in all normal circumstances the auth config will be hosted at the same place as the gRPC server, and it'll be at /auth, /ewb/auth, or /api/config/auth. We're in control of this through how we deploy, however there's always a possibility that we'll have to change the conf address path or host in the future, so it is worth supporting that now.